code/lab/wmcgui.cpp

```diff
@@ -2701,11 +2701,13 @@ float Slider::GetSliderPos(int x)
 
 void Slider::UpdateSlider(int x)
```

Comment:
I dislike how this function currently does two things at the same time which required you to add the `x >= 0` workaround.

Could you change it so that this function would be passed a `float` value which sets the `SliderScale` directly without calling `GetSliderPos`?

The existing `UpdateSlider(x)` invocations could be changed to `UpdateSlider(GetSliderPos(x))` to preserve the previous behaviour. This would lead to a much cleaner API and would generally be much easier to understand.